stream.js

```diff
@@ -27,7 +31,14 @@ function WebSocketStream(target, protocols, options) {
     socket = target
   // otherwise make a new one
   } else {
-    socket = new WS(target, protocols, options)
+    // special constructor treatment for native websockets in browsers, see
+    // https://github.com/maxogden/websocket-stream/issues/82
+    if (isNative && isBrowser && !protocols) {
+      socket = new WS(target)
```

Comment:
Wouldn't this fail in the case where you need to pass `protocols` to a native websockets impl? From some testing in safari it looks like any call w/ `arguments.length > 2` will throw a `TypeError`.

ISTM we could just drop the `!protocols` check from the first case -- something like this:

```js
if (isNative && isBrowser) {
  socket = new WS(target, protocols)
} else {
```

WFM at least.